spec/models/event_spec.rb

```diff
@@ -88,4 +88,50 @@
       end
     end
   end
+
+  describe ".search" do
```

Comment:
This is some careful testing so that's a good thing, but if we keep up with this standard, it'll become very hard to test something more complicated with more branches. Plus, having too many tests makes the test suite slow.

I suggest that you have only one context, where the keyword and the event names are carefully designed so that all the edges I listed are taken into account.